pkg/retagger/executor.go

```diff
@@ -21,7 +29,148 @@ func (r *Retagger) ExecuteJobs() error {
+		r.executeSingleJob(job, false) // Do not skip existing tags for this job
+	}
+
+	return nil
+}
+
+// compilePatternJobs gets tag lists from the upstream repo, finds tags matching the pattern specified in the job,
+// and creates new jobs to retag new images and images which have changed since their last collection.
+func (r *Retagger) compilePatternJobs(job Job) ([]Job, error) {
+	r.logger.Log("level", "debug", "message", fmt.Sprintf("compiling jobs for image %v using pattern %v, with options %#v", job.SourceImage, job.SourcePattern, job.Options))
+
+	// Make sure our pattern is valid
+	pattern, err := regexp.Compile(job.SourcePattern)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	// Get SHA/Tag pairs from our quay registry
+	quayTagMap, err := r.registry.GetQuayTagMap(job.SourceImage)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	// Handle remote, Docker Hub, and Docker library image path formats
+	registryPath, err := r.registry.GuessRegistryPath(job.SourceImage)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	// Create a reference to the external registry
+	o := dockerRegistry.Options{
+		Logf:          dockerRegistry.Quiet,
+		DoInitialPing: false,
+	}
+	externalRegistry, err := dockerRegistry.NewCustom(fmt.Sprintf("https://%s", registryPath.Hostname()), o)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	fullImageName := r.registry.GetRepositoryFromPath(registryPath)
+
+	// Get the tags for this image from the external registry
+	externalRegistryTags, err := externalRegistry.Tags(fullImageName)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	// Find tags matching our configured pattern
+	matches := []string{}
+	for _, t := range externalRegistryTags {
+		if pattern.MatchString(t) {
+			matches = append(matches, t)
+		}
+	}
+
+	if len(matches) == 0 {
+		r.logger.Log("level", "warn", "message", fmt.Sprintf("No upstream image tags were found matching the pattern %s", job.SourcePattern))
+	} else {
+		r.logger.Log("level", "debug", "message", fmt.Sprintf("Found %d upstream tags which match the pattern %s", len(matches), job.SourcePattern))
+	}
+
+	jobs := []Job{}
+	// Find tags which need to be re-checked and updated
+	for _, match := range matches {
+		sourceSHA := ""
+
+		_, exists := quayTagMap[match]
+
+		if !exists {
+			// Tag is new - get SHA and tag it
+			newDigest, err := externalRegistry.ManifestDigest(fullImageName, match)
+			if err != nil {
+				return nil, microerror.Mask(err)
+			}
+			sourceSHA = newDigest.String()
+
+		} else {
+			tag := quayTagMap[match]
+			if job.Options.UpdateOnChange {
+				// Tag exists, but we should update the image
+
+				newDigest, err := externalRegistry.ManifestDigest(fullImageName, tag.Name)
+				if err != nil {
+					return nil, microerror.Mask(err)
+				}
+
+				if newDigest.String() != tag.ManifestDigest {
+					// Retag this image with this tag
+					r.logger.Log("level", "debug", "message",
```

Comment:
I am not a big fan of splitting lines here. It would make more sense to me to have the `fmt` foo in its own line with a variable instead. 

Personal preference though.

Reply:
I un-split it. Looks nicer